Mbs 7969 k8soidc (#274)
avito-tech/avito-android: docs/content/docs/test/Runner.md

---
title: Test Runner
type: docs
---

> 

Reviewer: Add specific internal tag. @eugene-krivobokov help me, please.

Reviewer 2: `{{< avito >}}`

https://avito-tech.github.io/avito-android/docs/contributing/docs/#internal-links

Author: thanks, never used it

# Avito test runner

## How to run tests in kubernetes from a local machine (internal)

1. Get access to kubernetes cloud: [internal doc](http://links.k.avito.ru/Kubectl)
1. [Request](http://links.k.avito.ru/androidEmulatorServiceDesk) `exec` access to `android-emulator` namespace in `beta` cluster

> 1. Setup a context on `beta/android-emulator` with your user access (also described in first doc)

Reviewer: ```suggestion
1. Setup a context on `beta/android-emulator` with your user access. Direct path to section with context setup
```

1. Run tests with extra parameters specified, example for `:test-app`:

```shell script
./gradlew :subprojects:android-test:test-app:instrumentationUi 
    -Pci=true 

>     -PkubernetesContext=<your context>

Reviewer: It's not clear what context I need to type here.

Author: it is described in list above

Author: and you need to know about https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#define-clusters-users-and-contexts
beforehand

>     -PkubernetesCaCertFile=<kube cert>

Reviewer: Maybe we need an example path to file?

Author: wait what? ) like `/home/my.crt`? what could be unclear here?

Author: change to
`    -PkubernetesCaCertFile=<path to ca certificate file>`

Reviewer: I supposed that user can't understand what is a cacert file and where he can find it on local computer
